hapi-loader/src/test/java/com/arkhn/hapi/loader/EmbeddedKafkaIntegrationTest.java

```diff
@@ -0,0 +1,150 @@
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import org.springframework.test.context.ContextConfiguration;
+import org.testcontainers.elasticsearch.ElasticsearchContainer;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.jpa.api.dao.DaoRegistry;
+import ca.uhn.fhir.jpa.api.dao.IFhirResourceDao;
+import ca.uhn.fhir.jpa.api.model.DaoMethodOutcome;
+import ca.uhn.fhir.parser.IParser;
+
+import org.testcontainers.containers.PostgreSQLContainer;
+
+import com.arkhn.hapi.loader.embedded.KafkaProducer;
+
+@SpringBootTest(classes = ResourceConsumer.class)
+@EmbeddedKafka(partitions = 1, brokerProperties = { "listeners=PLAINTEXT://localhost:9092", "port=9092" })
+@ContextConfiguration(initializers = EmbeddedKafkaIntegrationTest.Initializer.class)
+class EmbeddedKafkaIntegrationTest {
+
+    private static ElasticsearchContainer embeddedElastic;
+    private static PostgreSQLContainer<?> embeddedPostgres;
+
+    @Autowired
+    public KafkaTemplate<String, String> template;
+
+    @Autowired
+    DaoRegistry daoRegistry;
+
+    @Autowired
+    FhirContext myFhirContext;
+
+    @Autowired
+    private KafkaProducer producer;
+
+    @Value("${test.topic}")
+    private String topic;
+
+    private static final String ELASTIC_VERSION = "7.10.1";
+    private static final String ELASTIC_IMAGE = "docker.elastic.co/elasticsearch/elasticsearch:" + ELASTIC_VERSION;
+
+    private static final String POSTGRES_IMAGE = "postgres:13";
+
+    @BeforeAll
+    public static void beforeClass() {
+        embeddedPostgres = new PostgreSQLContainer<>(POSTGRES_IMAGE).withDatabaseName("test-hapi-db").withUsername("sa")
+                .withPassword("sa");
+        embeddedPostgres.start();
+
+        embeddedElastic = new ElasticsearchContainer(ELASTIC_IMAGE)
+                .withStartupTimeout(Duration.of(300, ChronoUnit.SECONDS));
+        embeddedElastic.start();
+    }
+
+    @PreDestroy
+    public void stop() {
+        embeddedPostgres.stop();
+        embeddedElastic.stop();
+    }
+
+    @Test
+    public void loadNewResource() throws Exception {
+        // create a patient
+        Patient p = new Patient();
+        p.setId("loadNewResource");
+        p.setGender(AdministrativeGender.FEMALE);
+
+        // serialize it and send it to the loader
+        IParser parser = myFhirContext.newJsonParser();
+        String serialized = parser.encodeResourceToString(p);
+        producer.send(topic, serialized);
+
+        // wait a bit (could use a refactor)
+        Thread.sleep(3000);
+
+        // assert it has been created in the db
+        IFhirResourceDao<Patient> dao = daoRegistry.getResourceDao("Patient");
+        Patient result = dao.read(p.getIdElement());
+        assertNotNull(result);
```

Comment:
Should we also check that the fields in the doc are the right ones? (gender == female, etc.)

Reply:
will do